### PR TITLE
Issue 931  linearspace cmp

### PR DESCRIPTION
**Pull request by adler-j**
Fixes the first half of issue #931, does not add functionality for comparison.

### CODE REVIEW

File: odl/set/space.py [hunk 1]
```diff
@@ -803,6 +803,16 @@ def __pos__(self):
...
 
+    def __cmp__(self, other):
+        """Comparsion not implemented.
+
+        Raises
+        ------
+        TypeError
+        """
+        raise TypeError('unorderable type `{}`'
+                        ''.format(self.__class__.__name__))
```
Comment (kohr-h) [inline odl/set/space.py]: Can you try to mimic the upstream error message? It says, for example, `unorderable types: int() < str()`. Perhaps replace the comparison operator by a comma, i.e.
```python
raise TypeError('unorderable types: {}, {}'.format(self.__class__.__name__, type(other)))
```

**Comment by adler-j** [comment]
Added a more controversial change, related to comparsion of sets. Please have a look.

**Comment by kohr-h** [comment]
That's indeed more controversial. Do we want to have things like `grid <= partition`, `grid <= intv_prod` and the likes?

In some sense I feel that comparison operations `<=` should only compare within the same "category" while `contains_set` explicitly tells "compare in the sense of sets".
For instance, for grids, I think `<=`, if implemented at all, should invoke `is_subgrid` (not we have reversed logic here, we should go for one, either `is_subset` or `contains_grid`).

**Comment by adler-j** [comment]
Ok, I'll remove that part then and fix your comment, the merge.

**Comment by adler-j** [comment]
Also disallowed comparsion of `Set` instances in python 2 and added a test for it.

**Comment by kohr-h** [comment]
Okay, good fix.

**Pull request merged**